crabs/bboxes_labelling/combine_and_format_annotations.py

```diff
@@ -14,38 +15,44 @@
 @app.command()
 def combine_VIA_and_convert_to_COCO(
     parent_dir_via_jsons: str,
-    via_default_dir: str,
-    via_project_name: str,
+    exclude_pattern: Optional[str] = None,
+    via_default_dir: Optional[str] = None,
+    via_project_name: Optional[str] = None,
 ) -> str:
     """Combine a list of VIA JSON files into one and convert to COCO format
 
     Parameters
     ----------
     parent_dir_via_jsons : str
         path to the parent directory containing VIA JSON files
+    exclude_pattern : Optional[str], optional
+        a regex pattern that matches files to exclude.
+        By default, None.  E.g.: "\w+_coco_gen.json$"
     via_default_dir : str
-        The default directory in which to look for images for the VIA project.
-        A full path is required.
+        he default directory in which to look for images for the VIA project.
+        If None, the value specified in the first VIA JSON file is used.
+        If a path is provided it needs to be a full path.
     via_project_name : str
         The name of the VIA project.
+        If None, the value specified in the first VIA JSON file is used.
 
     Returns
     -------
     str
         path to the COCO json file. By default, the file
     """
 
-    # Get list of VIA JSON files
-    list_json_files = [
+    # Get list of all JSON files in directory
+    list_input_json_files = [
         x
         for x in Path(parent_dir_via_jsons).glob("*")
         if x.is_file() and str(x).endswith(".json")
     ]
```

Comment:
```suggestion
    all_files = Path(parent_dir_via_jsons).glob("*")
    is_a_json = lambda x: x.is_file() and str(x).endswith(".json")
    list_input_json_files = [x for x in all_files if is_a_json(x)]
```

Don't know if you agree but with a nicely named filter function and a nicely named variable, you can probably do away with the comment and get it all on one line.

Reply:
I like it, reads more easily than the first version. But I kept the conditional in the list comprehension because [ruff disapproves lambda assignments ](https://docs.astral.sh/ruff/rules/lambda-assignment/).